plugins/modules/digital_ocean_domain_record.py

```diff
@@ -0,0 +1,457 @@
+            # we have multiple similar records, bun not exact match
+            else:
+                # we have multiple similar records, can't decide what to do
+                if self.force_update:
+                    self.module.fail_json(msg="Can't update record, too many similar records: %s" % similar_records)
+                # create it
+                else:
+                    changed, result = self.__create_record()
+                return changed, result
+        # record matches
+        else:
+            changed = False
+            result = "Record has been already created"
+            return changed, result
+
+    def __update_record(self, record_id):
+        # before data comparison, we need to make sure that
+        # the payload['data'] is not normalized, but
+        # during create/update digitalocean expects normalized data
+        self.payload['data'] = self.__normalize_data()
+
+        # double check if the record exist
+        record = self.__find_record_by_id(record_id)
+
+        # record found
+        if record:
+            # PUT /v2/domains/$DOMAIN_NAME/records/$RECORD_ID
+            response = self.put('domains/%(domain)s/records/%(record_id)s' % {'domain': self.domain, 'record_id': record_id}, data=self.payload)
+            status_code = response.status_code
+            json = response.json
+            if status_code == 200:
+                changed = True
+                return changed, json['domain_record']
+            else:
+                self.module.fail_json(msg='Error updating domain record [%(status_code)s: %(json)s]' % {'status_code': status_code, 'json': json})
+        # recond not found
+        else:
+            self.module.fail_json(msg='Error updating domain record. Record does not exist. [%s]' % record_id)
+
+    def __build_payload(self):
+
+        payload = dict(
+            data=self.module.params.get('data'),
+            flags=self.module.params.get('flags'),
+            name=self.module.params.get('name'),
+            port=self.module.params.get('port'),
+            priority=self.module.params.get('priority'),
+            type=self.module.params.get('type'),
+            tag=self.module.params.get('tag'),
+            ttl=self.module.params.get('ttl'),
+            weight=self.module.params.get('weight')
+        )
+
+        # DigitalOcean stores every data in lowercase except TXT
+        if payload['type'] != 'TXT' and payload['data']:
+            payload['data'] = payload['data'].lower()
+
+        # digitalocean stores data: '@' if the data=domain
+        if payload['data'] == self.domain:
+            payload['data'] = '@'
+
+        return payload
+
+    def delete_record(self):
+
+        # if record_id is given, try to find the record based on the id
+        if self.record_id:
+            record = self.__find_record_by_id(self.record_id)
+            record_id = self.record_id
+        # if no record_id is given, try to find an exact match
```

Comment:
I'm using this already. Thanks. One thing is `state: present` matches records differently to `state: absent`. This means if you install a domain record then simply flip `state: present` to `state: absent` the module will not find the record you just created. `state: present` and `state: absent` need to use the same logic to locate an existing record.

Reply:
Suggested change: https://github.com/rezen/community.digitalocean/pull/3/files